src/graph/serializers/decoders/prev/v5/decode_graph.c

```diff
@@ -0,0 +1,166 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include <assert.h>
+#include "decode_v5.h"
+
+typedef enum {
+	V5_T_NULL = 0,
+	V5_T_STRING = 0x001,
+	V5_T_INT64 = 0x004,
+	V5_T_BOOL = 0x010, // shares 'longval' representation in SIValue union
+	V5_T_DOUBLE = 0x040,
+	V5_T_PTR = 0x080,
+	V5_T_CONSTSTRING = 0x100, // only used in deserialization routine
+	V5_T_NODE = 0x200,
+	V5_T_EDGE = 0x400,
+} PrevSIType;
+
+static SIType _ConvertSIType(PrevSIType prev_type) {
+	switch(prev_type) {
+	case V5_T_INT64:
+		return T_INT64;
+	case V5_T_DOUBLE:
+		return T_DOUBLE;
+	case V5_T_STRING:
+	case V5_T_CONSTSTRING:
+		return T_STRING;
+	case V5_T_BOOL:
+		return T_BOOL;
+	case V5_T_NULL:
+		return T_NULL;
+	default: // should not occur
+		assert(false);
+	}
+}
+
+static SIValue _RdbLoadSIValue_v5(RedisModuleIO *rdb) {
+	/* Format:
+	 * SIType
+	 * Value */
+	PrevSIType prev_type = RedisModule_LoadUnsigned(rdb);
+	SIType t = _ConvertSIType(prev_type); // Convert SIType to latest enum value
+
+	switch(t) {
+	case T_INT64:
+		return SI_LongVal(RedisModule_LoadSigned(rdb));
+	case T_DOUBLE:
+		return SI_DoubleVal(RedisModule_LoadDouble(rdb));
+	case T_STRING:
+		// Transfer ownership of the heap-allocated string to the
+		// newly-created SIValue
+		return SI_TransferStringVal(RedisModule_LoadStringBuffer(rdb, NULL));
+	case T_BOOL:
+		return SI_BoolVal(RedisModule_LoadSigned(rdb));
+	case T_NULL:
+	default: // currently impossible
+		return SI_NullVal();
```

Comment:
shouldn't be an error or feedback?

Reply:
We are restricted here to `printf` feedback. We could have an assertion failure here, but outside of the `_ConvertSIType` routine, none of this code is new - just moved from the decoding routines on master.